vllm/core/block/evictor.py

```diff
@@ -0,0 +1,110 @@
+import enum
+from abc import ABC, abstractmethod, abstractproperty
+from typing import OrderedDict
+
+
+class EvictionPolicy(enum.Enum):
+    """Enum for eviction policy used by make_evictor to instantiate the correct
+       Evictor subclass.
+    """
+    LRU = enum.auto()
+
+
+class Evictor(ABC):
+    """The Evictor subclasses should be used by the BlockAllocator class to
+    handle eviction of freed PhysicalTokenBlocks.
+    """
+
+    @abstractmethod
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def __contains__(self, content_hash: int) -> bool:
+        pass
+
+    @abstractmethod
+    def evict(self) -> int:
+        """Runs the eviction algorithm and returns the evicted block's
+        content hash
+        """
+        pass
+
+    @abstractmethod
+    def add(self, content_hash: int, num_hashed_tokens: int,
+            last_accessed: int):
+        """Adds block to the evictor, making it a candidate for eviction"""
+        pass
+
+    @abstractmethod
+    def remove(self, content_hash: int):
+        """remove block from the evictor of the same hash"""
+        pass
+
+    @abstractproperty
+    def num_blocks(self) -> int:
+        pass
+
+
+class Block():
```

Comment:
There's a bit of a name collision here. Can you rename this since "Block" is already used extensively? I went with BlockMetaData. Feel free to use that or a new name :). EvictorBlockInfo?

Reply:
BlockMetaData is nice, I'll take it~